samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -174,7 +174,7 @@ public ClusterBasedJobCoordinator(Config coordinatorSystemConfig) {
 
     // build a JobModelManager and ChangelogStreamManager and perform partition assignments.
     changelogStreamManager = new ChangelogStreamManager(coordinatorStreamManager);
-    jobModelManager = JobModelManager.apply(coordinatorStreamManager, changelogStreamManager.readPartitionMapping());
+    jobModelManager = JobModelManager.apply(coordinatorStreamManager.getConfig(), changelogStreamManager.readPartitionMapping());
```

Comment:
any reason to use a different constructor? would be nice to keep this PR just focussed on the KafkaConsumer and revert other changes

Reply:
This change came up during testing and seems like a good thing to keep.
CoordinatorStreamManager has its own life cycle, so unnecessary passing a reference to it into another class (with its own life cycle) is not desired.This way we can stop it here, without worrying the other reference.